### ISSUE TITLE
TBSync loosing Ressources

### ISSUE THREAD

**Issue by PotatoCarl** [jobisoft/TbSync#635]
## Your environment

TbSync version: Current
Thunderbird version: Suse Tumblweed

## Expected behavior
Syncing with account with all Ressources

## Actual behavior
Randomly TBSync looses Ressources (e.g. Calendar, Contacs). They are not displayed anymore.
Account has to be deleted and reconfigured.

## Steps to reproduce
Use TBSync for some time

**Comment by t0mcat1337** [comment]
Same for me.... suddenly after arbitary amounts of time ressources like contacts or calendars disappear. What helps is to deactivate and reactivate the affected account. 
After it had re-contacted the server "contacts" and "calendars" appear again.

Would like to provide a debug.log of tbsync but on every start of thunderbird it gets overwritten, so its hard to freeze the one when the error appears. Couldn't it simply be appended / rotated?

**Comment by gituser789** [comment]
Hi there,

i figured out this **happens after each thunderbird update**.
I am using four calendars (all on MS-exchange), but only the default-calendar is missing after each update. I dont use the contact syncing. This happens since more than one year, i figured out after each update to delete the TBsync account and add it again. This takes less than 1 minute, but doing this all 4 weeks is a bit frustrating.

**System**
 * Manjaro
 * Connected to MS active directory exchange-server

**Status after updating thunderbird and before fixing TBsync**
![image](https://user-images.githubusercontent.com/62549000/206390293-c750ec03-2a22-475e-b307-091e9a8a08d9.png)

**Status after fixing TBsync**
![image](https://user-images.githubusercontent.com/62549000/206390382-16fcb268-dd4b-4ce0-9f00-ec07ec0c60d8.png)

**Comment by gituser789** [comment]
Update:
To bring back the calendar it is enough to deactivate the synchronization account and then reactivate it.

A complete setup (delete and add again) is not necessary.

**Comment by skid9000** [comment]
I'm also seeing this problem on several installs when updating Thunderbird  EAS ressources just get deactivated in TbSync.

Is there a workaround to prevent having to reactivate ressources after each update ?

**Comment by daveknave** [comment]
I can confirm this faulty behavior as described above. However, I cannot observe:

1.  that this is bound to TB-updates being performed. Actually, this happens almost every day overnight, even without any update activity. I am using TB sync in Thunderbird on multiple Windows (10 and 11) and an Ubuntu (20.04) devices and this occurs on all of them.
2. I cannot observe any true randomness in calendars that go missing. Rather, all the calendars from one particular EAS account will disappear – solution as described above.

There is no conclusive information to be found in the debug logs. Here are the Log-Messages after booting for the first time after TBsync was last able to sync properly.

`
** Tue Feb 07 2023 09:12:33 GMT+0100 (Mitteleuropäische Normalzeit) **
[setSyncState] : State: prepare.request.options, Account: FU Exchange

** Tue Feb 07 2023 09:12:33 GMT+0100 (Mitteleuropäische Normalzeit) **
[Sending] : OPTIONS https://exchange.fu-berlin.de/Microsoft-Server-ActiveSync

** Tue Feb 07 2023 09:12:33 GMT+0100 (Mitteleuropäische Normalzeit) **
[setSyncState] : State: send.request.options, Account: FU Exchange

** Tue Feb 07 2023 09:12:33 GMT+0100 (Mitteleuropäische Normalzeit) **
[setSyncState] : State: eval.request.options, Account: FU Exchange

** Tue Feb 07 2023 09:12:33 GMT+0100 (Mitteleuropäische Normalzeit) **
[EAS OPTIONS with response (status: 200)] : 
responseText: 
responseHeader(MS-ASProtocolVersions): 2.0,2.1,2.5,12.0,12.1,14.0,14.1,16.0,16.1
responseHeader(MS-ASProtocolCommands): Sync,SendMail,SmartForward,SmartReply,GetAttachment,GetHierarchy,CreateCollection,DeleteCollection,MoveCollection,FolderSync,FolderCreate,FolderDelete,FolderUpdate,MoveItems,GetItemEstimate,MeetingResponse,Search,Settings,Ping,ItemOperations,Provision,ResolveRecipients,ValidateCert,Find

** Tue Feb 07 2023 09:12:33 GMT+0100 (Mitteleuropäische Normalzeit) **
[setSyncState] : State: prepare.request.folders, Account: FU Exchange

** Tue Feb 07 2023 09:12:33 GMT+0100 (Mitteleuropäische Normalzeit) **
[setSyncState] : State: send.request.folders, Account: FU Exchange

** Tue Feb 07 2023 09:12:33 GMT+0100 (Mitteleuropäische Normalzeit) **
[Sending (EAS v2.5)] : POST https://exchange.fu-berlin.de/Microsoft-Server-ActiveSync?Cmd=FolderSync&User=**************&DeviceType=TbSync&DeviceId=MZTBff7de7b145a10c019b35a3a43072
`

**Comment by Michael-Hennemann** [comment]
Hi, I am seeing the same issue.  

The address book disappears and isn't even in the tbsync "available resources"  list anymore. I have to disable and enable the EAS account in tbsync, then it comes back. 

Saw this issue with an EAS connection to a Kopano server and to a SoGo server with different users on different Win10 Clients. One time with an installed TB and once with a TB portable I use for testing. 

Also it just happened without any TB updates being applied. Just left TB open in my testlab client VM, came back the next day and the address book was gone. 

cheers
Michael

**Comment by Ch1llb0** [comment]
I can see the very same behaviour here as well. It sometimes looks like it might be connected to the same mail account (calendar und address book) being synced to Thunderbird sessions on three PCs. The resource - it's always the calendar here - disappears and only shows up as soon as I untick the account from being synced and tick the sync box again: suddenly the calendar is back again, but it has to be re-configured in TB (custom colour and naming has been forgotten).

**Comment by jobisoft** [comment]
This is very strange, and I have no idea why. If a single resource is removed (but not others), then the server seems to not include that resource for whatever reason during an update request. I did fix an issue in the latest update from a few days ago, where TbSync forgot the entire EAS provider being installed. But I do not think these issues are related.

**Comment by t0mcat1337** [comment]
@jobisoft 
How can we give you more infos about this issue for tracking down what's going wrong here?

As I already mentioned above, perhaps it could make sense to let the Plugin write some sort of debug.log, which is always appended and never gets deleted / overwritten. 
The problem with this log currently is that it is overwritten on each start of thunderbird which makes it very hard to log the problem, as - at a plus - it occurs arbitary / suddenly.
I could imagine such a log could clearify the behaviour and show the trigger which leads to this dissapearing of resources.

**Comment by Ch1llb0** [comment]
> The problem with this log currently is that it is overwritten on each start of thunderbird which makes it very hard to log the problem, as - at a plus - it occurs arbitary / suddenly.

Indeed! Sometimes, Thunderbird does start with the resource available as e.g. the calendar is still visible (discernable by the color I've assigned to it) in the daily preview. Only when the plugin seems to start up, the resource suddenly disappears.

**Comment by gituser789** [comment]
default calender lost today.
deactivating and activating the TBSync synchronisation resolves the issue.
